"Auto" doesn't work with opentelemetry-javaagent v0.9.0

I am trying to use opentelemetry javaagent and export traces to cloud trace. I am running into "No span exporter found" issues:

```
java -Dotel.exporter.jar=path/to/exporter-trace-0.9.1.jar  -javaagent:path/to/opentelemetry-javaagent-0.9.0-all.jar -jar path/to/myApp.jar
[opentelemetry.auto.trace 2020-11-04 08:08:19:881 +0530] [main] WARN io.opentelemetry.javaagent.tooling.TracerInstaller - No span exporter found in path/to/exporter-trace-0.9.1.jar
[opentelemetry.auto.trace 2020-11-04 08:08:19:882 +0530] [main] WARN io.opentelemetry.javaagent.tooling.TracerInstaller - No valid exporter found. Tracing will run but spans are dropped
[opentelemetry.auto.trace 2020-11-04 08:08:19:901 +0530] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 0.9.0
Hello world
[opentelemetry.auto.trace 2020-11-04 08:08:19:916 +0530] [opentelemetry-exec-shutdown-hook] WARN io.opentelemetry.javaagent.tooling.CommonTaskExecutor - Periodic task scheduler is shutdown. Will not run: cleaner for com.blogspot.mydailyjava.weaklockfree.WeakConcurrentMap@7d775612
```
Any suggestions on what could I be missing here or ways to investigate this further?

Hi @nilebox, thanks for you response. I tried with exporter-auto too, but I'm running into the same problem.
```
java -Dotel.exporter.jar=path/to/exporter-auto-0.9.1.jar  -javaagent:path/to/opentelemetry-javaagent-0.9.0-all.jar -jar path/to/myApp.jar
[opentelemetry.auto.trace 2020-11-04 09:06:32:199 +0530] [main] WARN io.opentelemetry.javaagent.tooling.TracerInstaller - No span exporter found in path/to/exporter-auto-0.9.1.jar
[opentelemetry.auto.trace 2020-11-04 09:06:32:200 +0530] [main] WARN io.opentelemetry.javaagent.tooling.TracerInstaller - No valid exporter found. Tracing will run but spans are dropped
[opentelemetry.auto.trace 2020-11-04 09:06:32:220 +0530] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 0.9.0
Hello world
[opentelemetry.auto.trace 2020-11-04 09:06:32:234 +0530] [opentelemetry-exec-shutdown-hook] WARN io.opentelemetry.javaagent.tooling.CommonTaskExecutor - Periodic task scheduler is shutdown. Will not run: cleaner for com.blogspot.mydailyjava.weaklockfree.WeakConcurrentMap@7d3ec58f
```
I also looked at https://github.com/GoogleCloudPlatform/opentelemetry-operations-java/blob/master/build.gradle to see if I am using the wrong versions. It seems we need to use 0.8.0 of javaagent with 0.9.1 of exporter-auto. However, that leads to `Caused by: java.lang.ClassNotFoundException: com.google.cloud.opentelemetry.trace.TraceExporter`
Any suggestions on this?

@nilebox Even after this change and on using the exporter-auto generated with it, I still get this:
```
Caused by: java.lang.NoClassDefFoundError: com/google/cloud/opentelemetry/trace/TraceExporter
	at com.google.cloud.opentelemetry.auto.GoogleCloudSpanExporterFactory.fromConfig(GoogleCloudSpanExporterFactory.java:15)
	at io.opentelemetry.javaagent.tooling.TracerInstaller.installExporter(TracerInstaller.java:181)
	at io.opentelemetry.javaagent.tooling.TracerInstaller.installExportersFromJar(TracerInstaller.java:155)
	at io.opentelemetry.javaagent.tooling.TracerInstaller.installAgentTracer(TracerInstaller.java:51)
	... 18 more
Caused by: java.lang.ClassNotFoundException: com.google.cloud.opentelemetry.trace.TraceExporter
	at io.opentelemetry.javaagent.tooling.ExporterClassLoader.findClass(ExporterClassLoader.java:80)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	... 22 more
```
Am I missing something more here that needs to be fixed?

@nilebox Looks like we need to supply exporter-trace as xbootclasspath for the agent to pick it up. However, it seems this too is not sufficient.

```
java -Xbootclasspath/a:path/to/opentelemetry-operations-java/exporters/trace/build/libs/exporter-trace-0.10.0-SNAPSHOT.jar -Dotel.exporter.jar=path/to/opentelemetry-operations-java-auto-fix/exporters/auto/build/libs/exporter-auto-0.10.0-SNAPSHOT.jar -javaagent:path/to/opentelemetry-javaagent-0.9.0-all.jar -jar path/to/myApp.jar

[opentelemetry.auto.trace 2020-11-05 11:23:02:018 +0530] [main] ERROR io.opentelemetry.javaagent.bootstrap.AgentInitializer - Throwable thrown while installing the agent tracer
java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.installAgentTracer(AgentInitializer.java:162)
	at io.opentelemetry.javaagent.bootstrap.AgentInitializer.initialize(AgentInitializer.java:75)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.agentmain(OpenTelemetryAgent.java:64)
	at io.opentelemetry.javaagent.OpenTelemetryAgent.premain(OpenTelemetryAgent.java:51)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:386)
	at sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:401)
Caused by: java.lang.NoClassDefFoundError: io/opentelemetry/sdk/trace/export/SpanExporter
	at java.lang.ClassLoader.findBootstrapClass(Native Method)
	at java.lang.ClassLoader.findBootstrapClassOrNull(ClassLoader.java:1008)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:407)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:405)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
	at com.google.cloud.opentelemetry.auto.GoogleCloudSpanExporterFactory.fromConfig(GoogleCloudSpanExporterFactory.java:15)
	at io.opentelemetry.javaagent.tooling.TracerInstaller.installExporter(TracerInstaller.java:181)
	at io.opentelemetry.javaagent.tooling.TracerInstaller.installExportersFromJar(TracerInstaller.java:155)
	at io.opentelemetry.javaagent.tooling.TracerInstaller.installAgentTracer(TracerInstaller.java:51)
	... 18 more
```

--- a/build.gradle
+++ b/build.gradle
@@ -24,7 +24,7 @@ subprojects {
         googleCloudVersion = '1.0.2'
         cloudMonitoringVersion = '2.0.1'
         openTelemetryVersion = '0.9.1'
-        openTelemetryInstrumentationVersion = '0.8.0'
+        openTelemetryInstrumentationVersion = '0.9.0'
         junitVersion = '4.13'
         mockitoVersion = '3.5.10'
         opentelemetryOperationsVersion = "0.9.1"  // CURRENT_RELEASE_VERSION
@@ -42,7 +42,7 @@ subprojects {
                 slf4j                        : "org.slf4j:slf4j-api:${slf4jVersion}",
                 opentelemetry_api            : "io.opentelemetry:opentelemetry-api:${openTelemetryVersion}",
                 opentelemetry_sdk            : "io.opentelemetry:opentelemetry-sdk:${openTelemetryVersion}",
-                opentelemetry_auto           : "io.opentelemetry.instrumentation.auto:opentelemetry-javaagent-tooling:${openTelemetryInstrumentationVersion}",
+                opentelemetry_auto           : "io.opentelemetry.javaagent:opentelemetry-javaagent-spi:${openTelemetryInstrumentationVersion}",
                 opentelemetry_operations_java: "com.google.cloud.opentelemetry:exporter-trace:${opentelemetryOperationsVersion}"
         ]
         testLibraries = [

--- a/exporters/auto/src/main/java/com/google/cloud/opentelemetry/auto/GoogleCloudMetricExporterFactory.java
+++ b/exporters/auto/src/main/java/com/google/cloud/opentelemetry/auto/GoogleCloudMetricExporterFactory.java
@@ -2,15 +2,15 @@ package com.google.cloud.opentelemetry.auto;
 
 import com.google.auto.service.AutoService;
 import com.google.cloud.opentelemetry.metric.MetricExporter;
-import io.opentelemetry.javaagent.tooling.exporter.ExporterConfig;
-import io.opentelemetry.javaagent.tooling.exporter.MetricExporterFactory;
+import io.opentelemetry.javaagent.spi.exporter.MetricExporterFactory;
 
 import java.io.IOException;
+import java.util.Properties;
 
 @AutoService(MetricExporterFactory.class)
 public class GoogleCloudMetricExporterFactory implements MetricExporterFactory {
     @Override
-    public MetricExporter fromConfig(ExporterConfig _config) {
+    public MetricExporter fromConfig(Properties _config) {
         try {
             return MetricExporter.createWithDefaultConfiguration();
         } catch (IOException ex) {

--- a/exporters/auto/src/main/java/com/google/cloud/opentelemetry/auto/GoogleCloudSpanExporterFactory.java
+++ b/exporters/auto/src/main/java/com/google/cloud/opentelemetry/auto/GoogleCloudSpanExporterFactory.java
@@ -2,15 +2,15 @@ package com.google.cloud.opentelemetry.auto;
 
 import com.google.auto.service.AutoService;
 import com.google.cloud.opentelemetry.trace.TraceExporter;
-import io.opentelemetry.javaagent.tooling.exporter.ExporterConfig;
-import io.opentelemetry.javaagent.tooling.exporter.SpanExporterFactory;
+import io.opentelemetry.javaagent.spi.exporter.SpanExporterFactory;
 import io.opentelemetry.sdk.trace.export.SpanExporter;
 import java.io.IOException;
+import java.util.Properties;
 
 @AutoService(SpanExporterFactory.class)
 public class GoogleCloudSpanExporterFactory implements SpanExporterFactory {
     @Override
-    public SpanExporter fromConfig(ExporterConfig _config) {
+    public SpanExporter fromConfig(Properties _config) {
         try {
             return TraceExporter.createWithDefaultConfiguration();
         } catch (IOException ex) {

